marketplaceItems/avatarCustomization101/appResources/appData/avatarCustomization101.js

```diff
@@ -0,0 +1,937 @@
+            ? newMaterialData.description
+            : ""; // default value
+
+        var materialMappingScale = newMaterialData.materialMappingScale 
+            ? newMaterialData.materialMappingScale 
+            : { x: 1, y: 1 }; // default value
+
+        var newMaterials = newMaterialData.materials;
+
+        // If material already exists and isNamed === false
+        // Merge old material properties and new material properties
+        if (materialID && !isNamed) {
+
+            var oldMaterialDataString;
+            var oldMaterials = {};
+
+            oldMaterialDataString = Entities.getEntityProperties(materialID, ["materialData"]).materialData;
+            try {
+                // get old materials values and properties to carry over to new
+                oldMaterials = JSON.parse(oldMaterialDataString).materials;
+            } catch (e) {
+                console.error("Issues parsing oldMaterialDataString" + e);
+            }
+
+            // merge new materials and old material properties together
+            newMaterials = mergeObjectProperties(oldMaterials, newMaterials);
+
+            print("ROBIN IS THIS " + JSON.stringify(newMaterials));
+
+            if (newMaterials["unlit"] && !isPBR) {
+                // unlit exists and is false
+                // is shadeless
+                delete newMaterials["unlit"];
+            }
+
+        }
+
+        // Create new material 
+        if (!materialID) {
+
+            newMaterials.model = "hifi_pbr"; 
+
+            // create material entity properties
+            materialEntityProperties = {
+                // static defaults
+                type: "Material",
+                name: "Avatar101-Material",
+                parentID: MyAvatar.sessionUUID,
+                materialURL: "materialData",
+                priority: 1,
+                parentMaterialName: 1,
+                description: description,
+                materialMappingScale: materialMappingScale,
+                materialData: JSON.stringify({
+                    materialVersion: 1, 
+                    materials: newMaterials 
+                })
+            };
+
+            materialID = Entities.addEntity(materialEntityProperties, "avatar");
+
+        } else {
+
+            // Update old material
+            var updates = {
+                description: description,
+                materialMappingScale: materialMappingScale,
+                materialData: JSON.stringify({
+                    materialVersion: 1, 
+                    materials: newMaterials 
+                })
+            };
+
+            Entities.editEntity(materialID, updates);
+        }
+
+        if (isNamed) {
+            // is Named
+            // Update the ui with new properties
+            setMaterialPropertiesToDefaults();
+            updateMaterialDynamicDataUI(newMaterialData, isPBR);
+        } 
+    }
+
+    function prepImageBackendToUI(file) {
```

Comment:
perhaps a better name for this would be something like `getFilenameFromURL()`

Reply:
On it!